beacon-chain/db/kv/validated_checkpoint.go

```diff
@@ -0,0 +1,47 @@
+package kv
+
+import (
+	"context"
+	"encoding/binary"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+	bolt "go.etcd.io/bbolt"
+	"go.opencensus.io/trace"
+)
+
+func (s *Store) LastValidatedCheckpoint(ctx context.Context) ([32]byte, types.Slot, error) {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.LastValidatedCheckpoint")
+	defer span.End()
+
+	var lastChkPoint [32]byte
+	var slot types.Slot
+	err := s.db.View(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(lastValidatedCheckpoint)
+		val := bkt.Get([]byte("lastChkPoint"))
+		if len(val) != 40 {
+			return errors.New("invalid checkpoint point")
```

Comment:
We are trying to not use errors.New but rather create a named error in errors.go